docs/required_libraries.md

```diff
@@ -0,0 +1,47 @@
+# Required libraries
+
+This is a list of known third-party libraries required by the Crystal compiler and the standard library.
+
+## Runtime dependencies
```

Comment:
I would suggest to call this `Core dependencies` because *runtime* is ambiguous.

However, I think it might actually be better to merge all stdlib dependencies into a single table and use some indicator for core dependencies. Maybe a separate column? This column could also hold information about the stdlib modules that use this dependency. But the current solution in prose text is also good.